app/controllers/posts_controller.rb

```diff
@@ -1,86 +1,61 @@
 class PostsController < ApplicationController
   before_action :authenticate_user,except: [:index,:show]
 
-
   before_action :find_post,except: [:index,:new,:create]
 
 
-
-
-  def find_post
-    @post = Post.find params[:id]
-
-
-  end
-
-
   def index
     @posts=Post.includes(:category).order("created_at DESC").page(params[:page])
   end
 
   def new
-
     @post=Post.new
-
-
   end
 
   def create
-    p=params.require(:post).permit([:title, :body, :category_id,:image])
-
-    @post=Post.new(p)
+    @post=Post.new post_params
     @post.user=current_user
     if @post.save
       redirect_to post_path(@post)
     else
       render :new
     end
-
   end
 
   def show
-    find_post
```

Comment:
Don't need find post inside the method because you're already calling the "before_action" in the beginning of the  controller